Source/TrackableEntities.Client/ModelBase.cs

```diff
@@ -37,15 +39,128 @@ public abstract class ModelBase<TModel> : INotifyPropertyChanged
         /// Fire PropertyChanged event.
         /// </summary>
         /// <typeparam name="TResult">Property return type</typeparam>
-        /// <typeparam name="TSpecificModel">Specific entity type</typeparam>
         /// <param name="property">Lambda expression for property</param>
-        protected virtual void NotifyPropertyChanged<TSpecificModel, TResult>
-            (Expression<Func<TSpecificModel, TResult>> property)
+        protected virtual void NotifyPropertyChanged<TResult>
+            (Expression<Func<TModel, TResult>> property)
         {
-            string propertyName = ((MemberExpression)property.Body).Member.Name;
-            if (PropertyChanged != null)
+            NotifyPropertyChanged<TResult>(property);
```

Comment:
ReSharper complains that line 46 is recursive on all paths. Can we just remove this overload of `NotifyPropertyChanged`?